### ISSUE TITLE
eq_op false positive

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust-clippy#1666]
```
fn main() {
    const C: usize = 18;
    assert!(C > 0 && C < 24);
}
```

```
warning: equal expressions as operands to `&&`
 --> src\main.rs:5:13
  |
3 |     assert!(C > 0 && C < 24);
  |             ^^^^^^^^^^^^^^^
  |
  = note: #[warn(eq_op)] on by default
  = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#eq_op
```

**Comment by oli-obk** [comment]
This one is hard, since the values of the expressions are equal

**Comment by mcarton** [comment]
Clippy probably sees `true && true` here. I must admit I'm not a huge fan of dynamic asserts for static expressions, but that all we've got ([however…](http://play.integer32.com/?gist=176eb2a9594eb1a2b3981423e046e9a0&version=undefined)). What does the original code looks like?

**Comment by leonardo-m** [comment]
> I must admit I'm not a huge fan of dynamic asserts for static expressions, but that all we've got (however…). What does the original code look like?

That assert_positive macro breaks apart if I don't use an uint constant:

`const C: u32 = 18;`

The original code comes from D language code, that has `static assert();`.

The program used static numeric values at the top of the module as configuration (as length of fixed-size arrays). The assert (or static assert) is handy to make sure those constant configuration values can be handled correctly by the program.

**Comment by oli-obk** [comment]
You can always adjust the macro to accept any book value by casting the book to usize.

I suggest to turn off lints for your crate which trigger on static values.

**Comment by leonardo-m** [comment]
> I suggest to turn off lints for your crate which trigger on static values.

So do you suggest me to close this bug report as invalid?

**Comment by oli-obk** [comment]
Not until we get static assert back ^^ then this lint should suggest to use static assert. I'll try to split the lint in two for now, so you can disable the const one in your code

**Comment by mikerite** [comment]
I can't reproduce this.
[Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=37a3d94aa8fa4da255d47d8d178b945c)
